src/ray/util/util.h

```diff
@@ -195,41 +196,21 @@ struct EnumClassHash {
 template <typename Key, typename T>
 using EnumUnorderedMap = std::unordered_map<Key, T, EnumClassHash>;
 
-namespace ray {
-namespace internal {
-inline __suppress_ubsan__("signed-integer-overflow") int64_t GenerateSeed() {
-  int64_t seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
-  // To increase the entropy, mix in a number of time samples instead of a single one.
-  // This avoids the possibility of duplicate seeds for many workers that start in
-  // close succession.
-  for (int i = 0; i < 128; i++) {
-    std::this_thread::sleep_for(std::chrono::microseconds(10));
-    seed += std::chrono::high_resolution_clock::now().time_since_epoch().count();
-  }
-  return seed;
-}
-}  // namespace internal
-}  // namespace ray
-
 /// A helper function to fill random bytes into the `data`.
 /// Warning: this is not fork-safe, we need to re-seed after that.
 template <typename T>
 void FillRandom(T *data) {
   RAY_CHECK(data != nullptr);
-  auto randomly_seeded_mersenne_twister = []() {
-    std::mt19937 seeded_engine(ray::internal::GenerateSeed());
-    return seeded_engine;
-  };
 
   // NOTE(pcm): The right way to do this is to have one std::mt19937 per
   // thread (using the thread_local keyword), but that's not supported on
   // older versions of macOS (see https://stackoverflow.com/a/29929949)
   static std::mutex random_engine_mutex;
   std::lock_guard<std::mutex> lock(random_engine_mutex);
```

Comment:
What is the purpose of this lock? Was it used to make sure that `GenerateSeed` works properly? Can we remove it now?

Reply:
This lock is protecting the static variable down below (now `absl::BitGen`). As the comment says, the better solution is to make the variable `thread_local`, but it used to not be supported with our macOS compiler toolchain. However, since then we have upgraded the toolchain, so it is worth trying it out again. I think we should first merge this PR in its current form and then try thread_local to minimize the chances of problems (this change is pretty safe, but the thread_local one could cause problems).